src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1596,1436 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+    {
+        DoCastSelf(SPELL_COSMETIC_FEELING_SICK, false);
+
+        _events.ScheduleEvent(EVENT_SEA_DOG_VOMIT, 5s, 20s);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_SEA_DOG_VOMIT:
+                    DoCastSelf(SPELL_SICK, false);
+                    _events.Repeat(25s, 50s);
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+private:
+    EventMap _events;
+};
+
+enum OrcSeaPup
+{
+    NPC_ORC_CRATE                               = 44915,
+
+    SPELL_SUMMON_ORC_CRATE                      = 83835,
+    SPELL_EJECT_ALL_PASSENGERS                  = 68576,
+    SPELL_ANIM_DEAD                             = 98190,
+    SPELL_DESPAWN_ALL_SUMMONS                   = 83840,
+
+    EVENT_ORC_PUP_JUST_SUMMONED                 = 1,
+    EVENT_ORC_PUP_REMOVE_PROTECTION             = 2,
+    EVENT_ORC_PUP_TALK                          = 3,
+    EVENT_ORC_PUP_DELIVER_CRATES                = 4,
+
+    TALK_ORC_PUP_SUMMONED                       = 0,
+    TALK_ORC_PUP_DELIVER_CRATES                 = 6,
+    TALK_ORC_PUP_WORN_OFF                       = 7
+};
+
+// 44914 - Orc Sea Pup
+struct npc_silverpine_orc_sea_pup : public VehicleAI
+{
+    npc_silverpine_orc_sea_pup(Creature* creature) : VehicleAI(creature), _isJustSummoned(true) { }
+
+    void JustAppeared() override
+    {
+        me->SetReactState(REACT_PASSIVE);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (Player* player = summoner->ToPlayer())
+        {
+            _playerGUID = player->GetGUID();
+
+            me->RemoveNpcFlag(UNIT_NPC_FLAG_SPELLCLICK);
+
+            me->GetMotionMaster()->MoveFollow(player, 2.0f, float(M_PI / 2.0f));
+
+            if (player->GetQuestStatus(QUEST_STEEL_THUNDER) == QUEST_STATUS_INCOMPLETE)
+            {
+                _events.ScheduleEvent(EVENT_ORC_PUP_JUST_SUMMONED, 1s);
+                _events.ScheduleEvent(EVENT_ORC_PUP_REMOVE_PROTECTION, 1s + 500ms);
+                _events.ScheduleEvent(EVENT_ORC_PUP_TALK, 5s);
+
+                int c = player->GetReqKillOrCastCurrentCount(QUEST_STEEL_THUNDER, NPC_ORC_CRATE);
+
+                for (int i = 0; i < c; i++)
+                    DoCastSelf(SPELL_SUMMON_ORC_CRATE);
+            }
+        }
+    }
+
+    void PassengerBoarded(Unit* passenger, int8 seatId, bool apply) override
+    {
+        if (passenger->GetEntry() == NPC_ORC_CRATE)
+        {
+            if (apply)
+            {
+                if (_isJustSummoned)
+                    return;
+
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    Talk(seatId + 1, player);
+            }
+            else
+                passenger->ToCreature()->DespawnOrUnsummon(3s);
```

Comment:
what ifr passenger is no creature? (ref gm magic)

Reply:
Added IsCreature() check to avoid any crash.